### PR TITLE
RELATED: FET-1109 Enzyme to RTL (ConfigurationControl)

### PR DESCRIPTION
**Pull request by marmil-cz**
Migrate tests to react-testing-library

JIRA: FET-1109

---

Supported PR commands:

| Command                  | Description             |
| ------------------------ | ----------------------- |
| `ok to test`             | Re-run standard checks  |
| `extended test`          | BackstopJS tests        |
| `extended check sonar`   | SonarQube tests         |
| `extended check cypress` | Cypress E2E tests       |
| `extended check plugins` | Dashboard plugins tests |

---

# PR Checklist

-   [ ] commit messages adhere to the [commit message guidelines](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#what-should-the-commits-look-like)
-   [ ] review was done by a Code owner [if necessary](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-tell-if-my-pull-request-needs-approval-by-a-code-owner) (if you think it is not necessary, explain the reasoning in the description or in a comment)
-   [ ] `check` passes
-   [ ] `check-extended` passes
-   [ ] `check-extended-cypress` passes
-   [ ] `rush change` [was run if applicable](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-describe-my-changes-for-the-changelog)

### CODE REVIEW

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/9241/ : SUCCESS in 9s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/11070/ : FAILURE in 9m 53s

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/9243/ : SUCCESS in 9s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/11073/ : FAILURE in 9m 48s

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by marmil-cz** [comment]
extended test

**Comment by marmil-cz** [comment]
extended check cypress

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by gdgate** [comment]
Build in pipeline check-extended-cypress aborted.

**Comment by marmil-cz** [comment]
extended test

**Comment by marmil-cz** [comment]
extended check cypress

**Comment by yenkins** [comment]
# ✨ Cypress test results ✨

## 🎉 Great news! Tests were successful.

|Status|Name|Tests|Failures|Time|
|:--:|:--|:--:|:--:|--:|
|✅|**attributeFilterButtonParentChild.spec.ts**|3|-|18.9400s|
|✅|**pivotTableAggregationsMenuAndAddTotal.spec.ts**|5|-|24.9960s|
|✅|**dashboard.spec.ts**|9|-|53.6510s|
|✅|**attributeFilterButton.spec.ts**|9|-|35.1490s|
|✅|**headlineChartDrilling.spec.ts**|2|-|8.4720s|
|✅|**dateFilter.spec.ts**|9|-|18.8640s|
|✅|**pivotTableSizingReset.spec.ts**|15|-|42.7000s|
|✅|**pivotTableRemoveAllFromMutipleAggregation.spec.ts**|1|-|12.6200s|
|✅|**pivotTableRemoveOneAggregation.spec.ts**|1|-|12.7260s|
|✅|**drilling.spec.ts**|2|-|20.3220s|
|✅|**attributeFilterConfig.spec.ts**|3|-|30.3890s|
|✅|**parentChildFiltersWithPlaceholders.spec.ts**|3|-|22.7540s|
|✅|**pivotTableSizing.spec.ts**|1|-|9.1790s|
|✅|**pivotTableRemoveOneFromMutipleAggregation.spec.ts**|1|-|11.6470s|
|✅|**dashboardFiltering.spec.ts**|2|-|25.0380s|
|✅|**barChartDrilling.spec.ts**|4|-|11.0150s|
|✅|**attributeFilterButtonWithSelection.spec.ts**|2|-|11.0730s|

You can find full report [here](https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-cypress-tests/276/).

**Comment by gdgate** [comment]
Build succeeded (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/9253/ : SUCCESS in 8s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-cypress-tests/276/ : SUCCESS in 14m 15s

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/9251/ : SUCCESS in 12s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/11080/ : FAILURE in 10m 19s

**Comment by yenkins** [comment]
Link to Storybook test result: https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-backstop-tests/1790/artifact/libs/sdk-ui-tests/backstop/output/html-report/index.html

**Comment by gdgate** [comment]
Build succeeded (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/9252/ : SUCCESS in 6s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-backstop-tests/1790/ : SUCCESS in 25m 42s

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/9254/ : SUCCESS in 16s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/11081/ : FAILURE in 9m 52s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/9258/ : SUCCESS in 8s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/11085/ : SUCCESS in 17m 06s

File: libs/sdk-ui-ext/src/internal/components/configurationControls/ConfigSection.tsx [hunk 1]
```diff
@@ -69,7 +69,7 @@ export class ConfigSection extends React.Component<IConfigSectionProps, IConfigS
         const configSectionClassName = `adi-bucket-configuration ${className}`;
 
         return (
-            <div className={this.getSectionClassNames()}>
+            <div className={this.getSectionClassNames()} aria-label="Configuration section">
```
Comment (no23reason) [inline libs/sdk-ui-ext/src/internal/components/configurationControls/ConfigSection.tsx:72]: While this fine for now, there really should be localized as well. But do not change it now :)

**Review by no23reason — APPROVED** [review]
LGTM

**Comment by yenkins** [comment]
### Sonar scan result


More detail, see in https://sonarqube-gate.intgdc.com/dashboard?id=gooddata-ui-sdk-gate-PR2846

To scan for vulnerabilities in dependencies and run unit tests (to get coverage report in sonar) please comment your PR with 'extended check sonar'.

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-fossa-scanning-zuul-el7/2620/ : SUCCESS in 1m 05s (non-voting)
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-merge-compliance-check/2920/ : SUCCESS in 21s
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-sonarqube-scan-zuul-gate/1752/ : SUCCESS in 16m 37s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/11087/ : SUCCESS in 15m 49s

**Pull request merged**